File: 771 - Jewels and Stones.cpp
#include <set>
#include <string>
using namespace std;

class Solution {
public:
    int numJewelsInStones(string jewels, string stones) {
        int c = 0;
        set<char> jew;
        for (int i = 0; i < jewels.size(); i++) {
            jew.insert(jewels[i]);
        }
        for (int i = 0; i < stones.size(); i++) {
            if (jew.find(stones[i]) != jew.end()) {
                c++;
            }
        }
    
        return c;
        
    }
};

File: 771 - Jewels and Stones.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        int s = 0;
        Set<Character> jew = new HashSet<>();
        for (char c: jewels.toCharArray()) {
            jew.add(c);
        }
        for (char c: stones.toCharArray()) {
            if (jew.contains(c)) {
                s++;
            }
        }
        return s;
    }
}